apps/condo/domains/notification/schema/RegisterPushTokenService.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Generated by `createservice notification.RegisterPushTokenService --type mutations`
+ */
+
+const get = require('lodash/get')
+
+const { GQLCustomSchema } = require('@core/keystone/schema')
+
+const { PushToken: PushTokenAPI } = require('@condo/domains/notification/utils/serverSchema')
+const access = require('@condo/domains/notification/access/RegisterPushTokenService')
+
+const RegisterPushTokenService = new GQLCustomSchema('RegisterPushTokenService', {
+    types: [
+        {
+            access: true,
+            type: 'input RegisterPushTokenInput { dv: Int!, sender: SenderFieldInput!, deviceId: String!, token: String!, serviceType: String! }',
+        },
+        {
+            access: true,
+            type: 'type RegisterPushTokenOutput { id: String!, deviceId: String!, token: String!, serviceType: String!, owner: User }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canRegisterPushToken,
+            schema: 'registerPushToken(data: RegisterPushTokenInput!): RegisterPushTokenOutput',
+            resolver: async (parent, args, context) => {
+                const { data: { dv, sender, deviceId, token, serviceType } } = args
+                const userId = get(context, 'authedItem.id', null)
+                const attrs = {
+                    dv,
+                    sender,
+                    deviceId,
+                    token,
+                    serviceType,
+                    owner: userId ? { disconnectAll: true, connect: { id: userId } } : null,
+                }
+                const existingToken = await PushTokenAPI.getOne(context, { deviceId, serviceType })
```

Comment:
If you want to create a tool for your case it's better to use something like `PushTokenAPI.updateOrCreate(context, { deviceId, serviceType }, attrs)`! @AntonAL look at this

Reply:
added `updateOrCreate` as we discussed yesterday evening.